src/OmniSharp.Roslyn.CSharp/Workers/Diagnostics/AnalyzerWorkQueue.cs

```diff
@@ -44,7 +44,7 @@ public AnalyzerWorkQueue(ILoggerFactory loggerFactory, int timeoutForPendingWork
             _maximumDelayWhenWaitingForResults = timeoutForPendingWorkMs;
         }
 
-        public void PutWork(IReadOnlyCollection<DocumentId> documentIds, AnalyzerWorkType workType)
+        public void PutWork(IReadOnlyCollection<Document> documents, AnalyzerWorkType workType)
```

Comment:
would this mean we would have stale documents in the queue potentially?

Reply:
Yep. That's really fine: Documents are immutable, so there's no threading concern. It could mean that we do a bit of extra work, but analysis on a single document or even group of documents isn't _that_ expensive, particularly since we don't have a UI thread dependency here.